controllers/apidefinition_controller.go

```diff
@@ -77,45 +78,60 @@ func (r *ApiDefinitionReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 
 	if desired.GetLabels()["template"] == "true" {
 		log.Info("Syncing template", "template", desired.Name)
+
 		res, err := r.syncTemplate(ctx, req.Namespace, desired)
 		if err != nil {
 			log.Error(err, "Failed to sync template")
 			return res, err
 		}
+
 		log.Info("Synced template", "template", desired.Name)
+
 		return ctrl.Result{}, nil
 	}
 
 	var queueA time.Duration
+
 	_, err = util.CreateOrUpdate(ctx, r.Client, desired, func() error {
 		if !desired.ObjectMeta.DeletionTimestamp.IsZero() {
 			e, err := r.delete(ctx, desired)
 			queueA = e
 			return err
 		}
+
 		if desired.Spec.APIID == "" {
 			desired.Spec.APIID = encodeNS(req.NamespacedName.String())
 		}
+
 		if desired.Spec.OrgID == "" {
 			desired.Spec.OrgID = env.Org
 		}
+
 		util.AddFinalizer(desired, keys.ApiDefFinalizerName)
-		for _, certName := range desired.Spec.CertificateSecretNames {
+
+		// we support only one certificate secret name for mvp
+		if len(desired.Spec.CertificateSecretNames) != 0 {
+			certName := desired.Spec.CertificateSecretNames[0]
```

Comment:
What's this change about? is it a bug?

Reply:
It was a for loop before which processed first record and then break unconditionally. Go linter was complaining about it.